src/analyzer.ts

```diff
@@ -259,32 +300,50 @@ export class StreamResolver implements Resolver {
 
   resolveDeferredFile(filePath: string, file: File): void {
     let deferred = this.deferredFiles.get(filePath);
-    deferred.resolve(file.contents.toString());
+    deferred(file.contents.toString());
     this.deferredFiles.delete(filePath);
   }
 
-  accept(url: string, deferred: Deferred<string>): boolean {
-    logger.debug(`accept: ${url}`);
+  canLoad(url: string): boolean {
+    return true;
+  }
+
+  load(url: string): Promise<string> {
+    logger.debug(`loading: ${url}`);
     let urlObject = parseUrl(url);
 
     // Resolve external files as empty strings. We filter these out later
     // in the analysis process to make sure they aren't included in the build.
     if (isDependencyExternal(url)) {
-      deferred.resolve('');
-      return true;
+      return Promise.resolve('');
     }
 
     let urlPath = decodeURIComponent(urlObject.pathname);
     let filePath = pathFromUrl(this.root, urlPath);
     let file = this.analyzer.getFile(filePath);
 
     if (file) {
-      deferred.resolve(file.contents.toString());
-    } else {
-      this.deferredFiles.set(filePath, deferred);
-      this.analyzer.pushDependency(urlPath);
+      return Promise.resolve(file.contents.toString());
     }
 
+    type Callback = (a: string) => string;
+    let callback: Callback;
+    const waitForFile = new Promise((resolve: Callback, reject: () => any) => {
+      callback = resolve;
+    });
+    this.deferredFiles.set(filePath, callback);
+    this.analyzer.pushDependency(urlPath);
+    return waitForFile;
+  }
+
+  /**
+   * Wraps the load() method to work in a way that is compliant with vulcanize
+   * & the old UrlResolver interface. To be removed once migration from
+   * hydrolosis to polymer-analyzer is complete.
+   */
+  accept(url: string, deferred: Deferred<string>): boolean {
+    this.load(url).then(deferred.resolve);
```

Comment:
maybe call `canLoad()` to ensure that the log is the same. (I know it's always true now)